aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -147,16 +147,15 @@ MPSDataType getMPSScalarType(ScalarType scalar_type) {
     return str;
 }
 
-MPSShape* getMPSShape(const Tensor& t) {
-  return getMPSShape(t.sizes());
+MPSShape* getMPSShape(const Tensor& t, c10::MemoryFormat memory_format) {
+  return getMPSShape(t.sizes(), memory_format);
 }
 
-MPSShape* getMPSShape(c10::MaybeOwned<Tensor> t) {
-  const Tensor& t_ = *t;
-  return getMPSShape(t_);
-}
-
-MPSShape* getMPSShape(IntArrayRef sizes) {
+MPSShape* getMPSShape(IntArrayRef sizes, c10::MemoryFormat memory_format) {
```

Comment:
This is same as the Conv shape calculation we do [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/mps/operations/Convolution.mm#L43). Please unify the logic.

Reply:
Also it will be better if you follow the Conv (N C H W ) channels naming, which makes it explicit and easy to read.

Reply:
Updated the PR.